### PR TITLE
[MM-56206] Allow for proper paging and sorting

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
When doing paging, I was not taking into account going through multiple previous pages. The correct way to do this was to define a `direction` for the cursor, such that we can scan rows up the table when going to previous pages using the first returned row, as well scanning down as we already supported.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56206

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/report.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
 		sortColumn = r.URL.Query().Get("sort_column")
 	}
 
+	direction := "down"
```
Comment (agnivade) [inline server/channels/api4/report.go:31]: We should probably move these to constants in the `web/params.go` file. Something for later.

**Review by agnivade — APPROVED** [review]
Just keep in mind that the client needs to ask in the reverse sort order as well when the direction is reversed. Otherwise, the results won't make sense.

**Review by harshilsharma63 — APPROVED** [review]
Looks good!

**Pull request merged**